app/src/main/java/org/sil/gatherwords/SessionActivity.java

```diff
@@ -57,31 +66,49 @@ protected void onCreate(Bundle savedInstanceState) {
         spinner.setAdapter(adapter);
         spinner.setOnItemSelectedListener(this);
 
+        // Find lable, speaker, eliciter EditTexts
+        labelField = findViewById(R.id.session_create_name);
+        speakerField = findViewById(R.id.session_create_speaker);
+        eliciterField = findViewById(R.id.session_create_eliciter);
+
         // Set date, time, and timezone fields
-        EditText dateField, timeField, timeZoneField;
         dateField = findViewById(R.id.session_create_date);
         timeField = findViewById(R.id.session_create_time);
         timeZoneField= findViewById(R.id.session_create_time_zone);
 
-        Date date = new Date();
-        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
-        dateField.setText(sdf.format(date));
-        sdf.applyPattern("HH:mm");
-        timeField.setText(sdf.format(date));
-        sdf.applyPattern("z");
-        String timeZoneString = sdf.format(date);
-        timeZoneField.setText(timeZoneString.substring(3, timeZoneString.length()));
+        dateSDF = new SimpleDateFormat("yyyy-MM-dd");
+        timeSDF = new SimpleDateFormat("HH:mm");
+        timeZoneSDF = new SimpleDateFormat("z");
+
+        Date date;
 
+        if ( creatingNewSession ) {
+            date = new Date();
+            dateField.setText(dateSDF.format(date));
+            timeField.setText(timeSDF.format(date));
+            String timeZoneString = timeZoneSDF.format(date);
+            timeZoneField.setText(timeZoneString.substring(3, timeZoneString.length()));
```

Comment:
What is the purpose of this substring operation?  Maybe leave a comment with what an example `timeZoneString` will look like.

`timeZoneString.substring(3, timeZoneString.length())` is equivalent to `timeZoneString.substring(3)`.